File: web3-node/send-eth-with-txn.js
const Web3 = require("web3");
var Tx = require("ethereumjs-tx").Transaction;

const rpcURL = process.env.INFURA_ROPSTEN_ACCESS_TOKEN;
const web3 = new Web3(rpcURL);

const address1 = process.env.ETH_ADDRESS_1;
const address2 = process.env.ETH_ADDRESS_2;

const privateKey1 = Buffer.from(process.env.ETH_PRIVATE_KEY_1, "hex");
const privateKey2 = Buffer.from(process.env.ETH_PRIVATE_KEY_2, "hex");

web3.eth.getTransactionCount(address1, (err, txCount) => {
  // Build the transaction
  const txObject = {
    nonce: web3.utils.toHex(txCount),
    to: address2,
    value: web3.utils.toHex(web3.utils.toWei("0.1", "ether")),
    gasLimit: web3.utils.toHex(21000),
    gasPrice: web3.utils.toHex(web3.utils.toWei("10", "gwei")),
  };

  // Sign the transaction
  const tx = new Tx(txObject, { chain: "ropsten" });
  tx.sign(privateKey1);

  const serializedTx = tx.serialize();
  const raw = "0x" + serializedTx.toString("hex");

  // Broadcast the transaction
  web3.eth.sendSignedTransaction(raw, (err, txHash) => {
    console.log("err: ", err);
    console.log("txHash:", txHash);
    // Now go check etherscan to see the transaction!
  });
});

File: python-backend/send-eth-with-txn.py
from web3 import Web3
from constants import *

from web3.middleware import construct_sign_and_send_raw_middleware

web3 = Web3(Web3.HTTPProvider(INFURA_ROPSTEN_ENDPOINT))
account = web3.eth.account.from_key(PRIVATE_KEY_STR_1)
web3.middleware_onion.add(construct_sign_and_send_raw_middleware(account))

value = web3.toWei("0.05", "ether")
gas = web3.eth.estimate_gas(
    {"value": value, "from": ADDRESS_1, "to": ADDRESS_2}
)
gas_price = web3.eth.gas_price

tx_hash = web3.eth.sendTransaction(
    {
        "from": ADDRESS_1,
        "to": ADDRESS_2,
        "value": value,
        "gas": gas,
        "gasPrice": gas_price,
    }
)

print(tx_hash.hex())
